src/components/modals/FlipInputModal2.tsx

```diff
@@ -0,0 +1,301 @@
+/* eslint-disable react-native/no-raw-text */
+
+import { div, log10, toFixed } from 'biggystring'
+import * as React from 'react'
+import { memo, useState } from 'react'
+import { TouchableWithoutFeedback, View } from 'react-native'
+import { AirshipBridge } from 'react-native-airship'
+import FontAwesomeIcon from 'react-native-vector-icons/FontAwesome'
+import { sprintf } from 'sprintf-js'
+
+import { MINIMUM_DEVICE_HEIGHT } from '../../constants/constantSettings'
+import { getSpecialCurrencyInfo } from '../../constants/WalletAndCurrencyConstants'
+import { useDisplayDenom } from '../../hooks/useDisplayDenom'
+import { useHandler } from '../../hooks/useHandler'
+import { useWatch } from '../../hooks/useWatch'
+import { formatNumber } from '../../locales/intl'
+import s from '../../locales/strings'
+import { deviceHeight } from '../../theme/variables/platform'
+import { useSelector } from '../../types/reactRedux'
+import { getCurrencyCode } from '../../util/CurrencyInfoHelpers'
+import { DECIMAL_PRECISION } from '../../util/utils'
+import { Card } from '../cards/Card'
+import { ExchangeRate2 } from '../common/ExchangeRate2'
+import { cacheStyles, Theme, useTheme } from '../services/ThemeContext'
+import { FiatText } from '../text/FiatText'
+import { EdgeText } from '../themed/EdgeText'
+import { ExchangedFlipInput2, ExchangedFlipInputAmounts, ExchangedFlipInputRef, ExchangeFlipInputFields } from '../themed/ExchangedFlipInput2'
+import { MiniButton } from '../themed/MiniButton'
+import { ThemedModal } from '../themed/ThemedModal'
+
+export type FlipInputModalResult = {
+  nativeAmount: string
+  exchangeAmount: string
+}
+
+export interface SetFeesParams {
+  feeTokenId?: string
+  feeNativeAmount: string
+}
+export interface FlipInputModalRef {
+  setFees: (params: SetFeesParams) => void
+  setAmount: (field: ExchangeFlipInputFields, value: string) => void
+  setError: (errorMessage: string | null) => void
+}
+
+type FeeStyleTypes = 'dangerText' | 'warningText'
+
+type Props = {
+  bridge: AirshipBridge<FlipInputModalResult>
+  walletId: string
```

Comment:
Please pass the wallet directly, as `wallet: EdgeCurrencyWallet`, instead of by ID.

Reply:
done